broker/data-access-layer/bosh/test/bosh.BoshDirectorClient.spec.js

```diff
@@ -1325,10 +1325,14 @@ describe('bosh', () => {
 
         dummyBoshDirectorClient.makeRequestWithConfig({}, 200, directorConfig)
           .then(() => {
-            expect(requestStub).to.be.calledOnce;
+            expect(requestStub).to.be.calledTwice;
             sandbox.restore();
```

Comment:
From the release 3.4.6 of bluebird they have fixed the issue of `Promise.map`. Earlier, it was performing the work synchronously whereas it was supposed to perform it asynchronously. Here, we are using `Promise.map` in the constructor of `BoshDirectorClient` and then we are creating the stubs. In the constructor itself, we are sending the `request` which was going to its actual function in earlier version of bluebird(3.4.0) but now as it is done asynchronously, the first call to `request` method is handled by stubs.